theano/gpuarray/dnn.py

```diff
@@ -298,18 +298,18 @@ def __init__(self, border_mode, conv_mode='conv',
     def make_node(self, kern_shape, subsample=(1, 1, 0)):
         if kern_shape.type.ndim != 1 or kern_shape.type.dtype != 'int64':
             raise TypeError('kern must be 1D shape tensor')
-        if isinstance(border_mode, integer_types):
-            border_mode = (border_mode,) * len(subsample)
-        if isinstance(border_mode, tuple):
-            assert len(border_mode) == len(subsample)
-            border_mode = tuple(map(int, border_mode))
-        if not ((isinstance(border_mode, tuple) and min(border_mode) >= 0) or
-                border_mode in ('valid', 'full', 'half')):
+        if isinstance(self.border_mode, integer_types):
```

Comment:
yes

On Mon, Jun 20, 2016 at 11:04 AM, Gokula Krishnan <notifications@github.com>
wrote:

> In theano/gpuarray/dnn.py
> <https://github.com/Theano/Theano/pull/4638#discussion_r67705850>:
>
> > @@ -298,18 +298,18 @@ def __init__(self, border_mode, conv_mode='conv',
> >      def make_node(self, kern_shape, subsample=(1, 1, 0)):
> >          if kern_shape.type.ndim != 1 or kern_shape.type.dtype != 'int64':
> >              raise TypeError('kern must be 1D shape tensor')
> > -        if isinstance(border_mode, integer_types):
> > -            border_mode = (border_mode,) * len(subsample)
> > -        if isinstance(border_mode, tuple):
> > -            assert len(border_mode) == len(subsample)
> > -            border_mode = tuple(map(int, border_mode))
> > -        if not ((isinstance(border_mode, tuple) and min(border_mode) >= 0) or
> > -                border_mode in ('valid', 'full', 'half')):
> > +        if isinstance(self.border_mode, integer_types):
>
> will do,
> shall I do the same with the other args as well?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/Theano/Theano/pull/4638/files/06f4ee27849eccfcc8649fb42b19d2082477e70e..ef387edc88f2f195f2318ed478f7ef03f71dbc42#r67705850>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AALC-w_Klmz5UFKX_o_hXyMpqVNkVDniks5qNqwCgaJpZM4I3m6N>
> .
>